### DISCUSSION TITLE
what is the best practice to implement this tab layout

### DISCUSSION THREAD

**Discussion by ShariatPanah** [dotnet/maui#20219] [Q&A]
so i'm curious what is the best practice to implement such tab layout? i'm not using Shell in this application so it's not an option for me, currently the tab buttons in this demo are defined using a Horizontal CollectionView but i don't know how to load the body of each Tab is being selected, is there a TabView Control that i can define the tab buttons the way i want? in Xamarin we had Tabview Control which was very flexible custom-wise but it's not yet ported to Maui

one option that i'm thinking about is to clear the body every time a tab is selected and load the corresponding details of the selected tab into the body using code behind, but this approach has a down side which is complication of code because every tab has it's own ViewModel and Logic.
so what should i do?

here is the demo i wanna implement:
![Tabs](https://github.com/dotnet/maui/assets/32509267/66197123-f980-4fe7-a538-ddfc3553a83c)

**Comment by Vroomer** [comment]
I suggest to take a look at Sharpnado.Tabs https://github.com/roubachof/Sharpnado.Tabs. Alternatively you could use CarouselView with custom template for indicators. In that case you can navigate between views by swiping/scrolling as well (which might not be desirable).
